src/cata_tiles.cpp

```diff
@@ -1654,9 +1655,18 @@ const tile_type *cata_tiles::find_tile_with_season( std::string &id )
     constexpr char season_suffix[4][suffix_len] = {
         "_season_spring", "_season_summer", "_season_autumn", "_season_winter"
     };
-
-    std::string seasonal_id = id + season_suffix[season_of_year( calendar::turn )];
-
+    int seasonnum = season_of_year( calendar::turn );
+    if( ( seasonnum == 3 ) && ( id == "t_grass" || id == "t_grass_dead" || id == "t_grass_long" ||
+                                id == "t_dirt" || id == "t_dirtmound" ) ) {
+        seasonnum = 2;
+    } else if( seasonnum == 3 && map::snowfall < 10000 ) {
+        seasonnum = 2;
+    }
+    if( seasonnum == 2 && map::snowfall > 10000 && !( id == "t_grass" || id == "t_grass_dead" ||
+            id == "t_grass_long" || id == "t_dirt" || id == "t_dirtmound" ) ) {
+        seasonnum = 3;
+    }
```

Comment:
Can the whole snippet be considered a temporary hack? The proper way of doing this is by means of the tileset jsons. Meanwhile, it deserves a thorough comment describing why it's here and under what circumstances it can be removed.

Reply:
This whole thing with the tilesets I'm gonna need to completely re-work, find a much better solution.
The winter versions of trees dont have visible fruit for example. 
And it needs to work for tilesets that don't have winter variants either.
I'll come back to this when I think of a better way.